### PR TITLE
chore: Add flag for disabling read tx timeout on BlockchainProvider

### LINKED ISSUE #11760: Add flag for disabling read tx timeout on `BlockchainProvider`
We need a flag on the legacy blockchain provider that will disable the database transaction timeout whenever a transaction is created. This should cause it to call the `disable_long_read_transaction_safety` method whenever new db providers are created, like we do for state root in the old engine:
https://github.com/paradigmxyz/reth/blob/47efb604ede2487c722b7af6ba567686b171068a/crates/blockchain-tree/src/blockchain_tree.rs#L1240

This should allow long running RPCs to be called without timing out when the legacy engine is used.

### PR DESCRIPTION
**Pull request by malik672**
closes #11760

### CODE REVIEW

File: crates/blockchain-tree/src/externals.rs [hunk 1]
```diff
@@ -38,8 +40,38 @@ impl<N: ProviderNodeTypes, E> TreeExternals<N, E> {
...
+{
+    pub fn get_provider(
+        &self,
+    ) -> DatabaseProvider<<<N as NodeTypesWithDB>::DB as Database>::TX, <N as NodeTypes>::ChainSpec>
+    {
+        let mut provider = self.provider_factory.provider().unwrap();
+        if self.disable_transaction_timeout {
+            provider = provider.disable_long_read_transaction_safety();
+        }
+        provider
```
Comment (Rjected) [inline crates/blockchain-tree/src/externals.rs:62]: we should not have `unwrap`s

**Review by Rjected — CHANGES_REQUESTED** [review]
I don't think adding a flag to the externals this way will work, because we still do `self.externals.provider_factory.provider()` in a lot of places in the legacy tree

File: crates/blockchain-tree/src/externals.rs (hunk 1 above)
Comment (malik672) [reply crates/blockchain-tree/src/externals.rs:62]: yeah, this should be more of a draft honestly, planned to remove the unwraps, just forgot

**Review by mattsse — COMMENTED** [review]
I dont want to change anything in this code because we're going to deprecate this soon

**Comment by malik672** [comment]
> I dont want to change anything in this code because we're going to deprecate this soon

I should leave it ?

**Pull request closed without merging**